src/material/tabs/tab-nav-bar/tab-nav-bar.ts

```diff
@@ -238,6 +259,9 @@ export class _MatTabLinkBase
     );
   }
 
+  /** Unique id for the tab. Provided by the sub-class. */
+  id: string;
```

Comment:
You can set this as `@Input` here in base class so you don't have to override it in the subclasses.

Reply:
Even if I add `@Input` here, I still need to override it in the sub classes because the non-MDC and MDC tabs need a differently formatted unique ID (one of the IDs has `mdc` in it, the other does not). I'm following the pattern of other components, but happy to do things differently.

Reply:
The ID pattern doesn't really matter, as long as it's unique. We should be able to avoid ID collisions if we only use the counter in the file with the base class.

Reply:
SGTM, updated. Also updated the non-MDC and MDC panels to use the same ID prefix.